pkg/boot/boottest/json.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package boottest
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/u-root/u-root/pkg/boot"
+	"github.com/u-root/u-root/pkg/curl"
+)
+
+func module(r io.ReaderAt) map[string]interface{} {
```

Comment:
Could this instead be an interface function. I seems a bit inflexible with the type checks.

Reply:
I'd rather not, (a) so it doesn't bloat code size -- this package is only used for tests, and (b) so nobody gets the idea to use this terrible format in production somewhere.

Reply:
do you want me to leave a comment in the code for this?

Reply:
Left a comment in the pkg comment.